crates/shared/src/bad_token/token_owner_finder.rs

```diff
@@ -34,6 +38,38 @@ pub trait TokenOwnerFinding: Send + Sync {
     async fn find_owner(&self, token: H160, min_balance: U256) -> Result<Option<(H160, U256)>>;
 }
 
+/// Arguments related to the token owner finder.
+#[derive(clap::Parser)]
+pub struct Arguments {
+    /// The token owner finding strategies to use.
+    #[clap(long, env, use_value_delimiter = true, arg_enum)]
+    pub token_owner_finders: Option<Vec<TokenOwnerFindingStrategy>>,
+
+    /// The fee value strategy to use for locating Uniswap V3 pools as token holders for bad token
+    /// detection.
+    #[clap(long, env, default_value = "static", arg_enum)]
+    pub token_owner_finder_uniswap_v3_fee_values: FeeValues,
+
+    /// Token owner finder-specific timeout configuration.
+    ///
+    /// This is a separate configuration flag because token holder APIs can sometimes take a long
+    /// time to complete (Blockscout will take up to 30 seconds sometimes). We still want these
+    /// longer requests to finish as bad token detection results are cached for a while and faster
+    /// token proposers aren't slowed down by slower ones.
+    #[clap(
+        long,
+        default_value = "45",
+        parse(try_from_str = duration_from_seconds),
+    )]
+    pub token_owner_finder_http_timeout: Duration,
+}
+
+impl Arguments {
+    fn http_client(&self) -> reqwest::Client {
+        crate::http_client(self.token_owner_finder_http_timeout)
+    }
+}
```

Comment:
I think we should still use the global http client because it might have been configured in other ways and generally is more efficient to reuse the same client. Also we only want this timeout for blockscout not for other implementations.

Reply:
Ok, fair point. I will change it back to passing in a client and then re-configuring it with a slightly different timeout.

Reply:
Looking at the code, this effectively means that we can't have multiple clients with different configurations and need to result to seeing different configurations per request (like setting the timeout per request) instead of configuring clients differently for different APIs.

While, in general I agree that having a shared client makes sense (reusing global configuration). However, in this case we are:
- Creating the Blockscout client with the same configuration (since we are reusing the `crate::http_client` function for this, and AFAICT, client configuration cannot be changed after creation.
- Would have to, as we need slightly different configuration options per API, need to add fields to the API implementations and set them per request, which is slightly error prone and less ergonomic (for example, setting authentication headers on every request instead of setting 

I'm going to undo this change for now (since it is a larger change), but I think the "correct" solution would be to introduce a:
```rust
struct HttpClientFactory { /* ... */ }

impl HttpClientFactory {
     /// Creates a returns a new client.
     pub fn create() -> reqwest::Client { /* ... */ }
     
     /// Returns a handle to the shared global client.
     pub fn shared() -> &reqwest::Client { /* ... */ }
     
     /// Creates a new
     pub fn configure(config: impl FnOnce(&mut reqwest::ClientBuilder)) -> reqwest::Client { /* ... */ }
}
```

I'm not 100% convinced either that sharing the client across separate hosts is beneficial. The only mention of client reuse I could find was:
> taking advantage of keep-alive connection pooling.

This only makes sense if you are reusing clients for the same host, as you can't `Connection: Keep-Alive` across hosts anyway. If this is the case, then it would make sense for each API to have a separate `Client` instance (guaranteed to be configured in some standard way by `HttpClientFactory`) and with their own connection pools.

Reply:
I wouldn't overthink it either. The only change we need at the moment is a longer timeout for blockscout. The simplest way to do that is to add it to the request. I feel like this machinery is overkill at the moment.

Reply:
I would argue that we should start using separate `reqwest::CLient` instances for our different API instances so they have separate `keep-alive` pools - since connections to different hosts can't be reused anyway.

Also, I don't think its overly complex - its basically `client_factory.create()` instead of `client.clone()` everywhere.

Reply:
Either way, this is a separate change, so I will undo it from this PR.

Reply:
Also, thinking about this a bit more, this adds a bit of a code-smell, for example:
```rust
BlockscoutDetector::try_with_network(
    Client::builder().timeout(Duration::from_secs(5)).build(),
    chain_id,
);
```

This will, unexpectedly, make HTTP requests with 45-second timeout instead of one with a 5 second timeout as the caller might expect.

Its not a big deal, but I do think it is a bit of limitation that we require a single `reqwest::Client` instead of using a factory.